csrc/type_promotion.h

```diff
@@ -13,20 +13,9 @@
 
 namespace nvfuser {
 
-//!
-//! The TypePromotionConfig flags are derived from Aten/TensorIterator.h
-//!
-//! 1) check_all_same_dtype_ flag checks that all inputs and defined outputs
-//! have the same dtype. Default = False
-//!
-//! 2) promote_inputs_to_common_dtype flag will cast the inputs to the common
-//! dtype. Default = True
-//!
-//! 3) promote_integer_inputs_to_float flag will cast the common dtype to the
-//! default float scalar type if it is an integral type (including bool).
-//!
```

Comment:
Was this comment removed because it's out of date? Maybe we can just update it instead.

Reply:
> Was this comment removed because it's out of date?

Yes. 

> Maybe we can just update it instead.

It's no longer fair to say the flags here are derived from ATen/TensorIterator.h. 
promote_integer_inputs_to_float comes from ATen/TensorIterator.h and require_full_precision_promoted doesn't. We also miss [a dozen of flags there](https://github.com/pytorch/pytorch/blob/242e03ba8661a62a1cf54ac89bda555b2120a6d4/aten/src/ATen/TensorIterator.h#L951-L961) if our intention is to match.